File: TekupPulse/src/main/java/tek_up/tekuppulse/ECSR/User/DTO/UserDTO.java
package tek_up.tekuppulse.ECSR.User.DTO;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import tek_up.tekuppulse.ECSR.User.Role;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder

public class UserDTO {
    private Long id;
    private String firstName;
    private String lastName;
    private String email;
    private Role role;
    private String profilePictureUrl;
    private String lastLoginIp;
    private LocalDateTime lastLoginTime;
    private Long groupId;
}

File: client/src/app/features/user/DTO/UserDTO.ts
export interface UserDTO {
  id?: number;
  firstName: string;
  lastName: string;
  email: string;
  password?: string; 
  role: 'STUDENT' | 'PROFESSOR' | 'ADMIN' | string;
  profilePictureUrl?: string;
  lastLoginIp?: string;
  lastLoginTime?: string;
  createdAt?: string;   
  groupId?: number;  
}
